pkg/graveler/committed/diff_test.go

```diff
@@ -238,6 +275,98 @@ func TestDiffCancelContext(t *testing.T) {
 		t.Fatalf("Err() returned %v, should return context.Canceled", err)
 	}
 }
+func TestNextRange(t *testing.T) {
+	ctx := context.Background()
+	it := committed.NewDiffIterator(ctx,
+		newFakeMetaRangeIterator([][]string{{"k1", "k2"}, {"k3", "k4"}}, [][]string{{"i1", "i2"}, {"i3", "i4"}}),
+		newFakeMetaRangeIterator([][]string{{"k3", "k4"}, {"k5", "k6"}}, [][]string{{"i3a", "i4a"}, {"i5", "i6"}}))
+
+	t.Run("next range - in middle of range", func(t *testing.T) {
+		if !it.Next() { // move to range k1-k2
+			t.Fatal("expected iterator to have value")
+		}
+		record, rng := it.Value()
+		if record != nil {
+			t.Fatal("expected record to be nil")
```

Comment:
Why not an error here?  Continuing the test would give more error messages, possibly useful.  Also for _many_ (but not _all_) of the fatals downstream.